### DISCUSSION TITLE
TShockAutoRegister plugin not working

### DISCUSSION THREAD

**Discussion by dringdahl** [Pryaxis/TShock#2352] [Q&A]
@moisterrific I apologize for asking here, but I didn't know where else to point out an issue, since there is no issues tab on your repository, and I assume the parent is dead or you would not have taken it over. This plugin is not working in the current version of TShock. It seems the API has been updated, but the plugin has not. I'm not a C# guy, but I might be able to fumble through it eventually if I could set up the environment to build the plugin. I just figured you'd probably be able to knock it out in short order.

As a side note, every plugin I tried from the list linked from the TShock README.md, except AdditionalPylons, did not work.

**Comment by moisterrific** [comment]
Thanks for letting me know, I'll try to fix it. TShockAPI recently made a change that broke a bunch of plugins, the fix is quite simple, just need to update the part that reference TShock Configuration and add `Settings`

**Reply by moisterrific** [reply]
I've updated the plugin and tested to make sure it works.

https://github.com/moisterrific/TShockAutoRegister

You should get the source code and try to build it if you have some experience with programming.

**Reply by dringdahl** [reply]
I got this when I tested it:

`127.0.0.1:49624 is connecting...`
`Debug: GetDataHandlers / HandlePlayerBuffList handled event and sent data Bobby`
`[Server API] Warning Plugin "AutoRegister" has had an unhandled exception thrown by one of its ServerJoin handlers:`
`System.NullReferenceException: Object reference not set to an instance of an object.`
`   at AutoRegister.Plugin.OnServerJoin(JoinEventArgs args)`
`   at TerrariaApi.Server.HandlerCollection`1.Invoke(ArgsType args) in `D:\a\TShock\TShock\TerrariaServerAPI\TerrariaServerAPI\TerrariaApi.Server\HandlerCollection.cs:line 109`
`Debug: GetDataHandlers / HandleSpawn force teleport 'vanilla spawn' Bobby`
`[Server API] Warning Plugin "AutoRegister" has had an unhandled exception thrown by one of its NetGreetPlayer handlers:`
`System.NullReferenceException: Object reference not set to an instance of an object.`
`   at AutoRegister.Plugin.OnGreetPlayer(GreetPlayerEventArgs args)`
`   at TerrariaApi.Server.HandlerCollection'1.Invoke(ArgsType args) in `
`D:\a\TShock\TShock\TerrariaServerAPI\TerrariaServerAPI\TerrariaApi.Server\HandlerCollection.cs:line 109`
`Bobby has joined.`

And in game it did not give any message about registering an account for "Bobby"

**Reply by dringdahl** [reply]
Seems the above problem was due to an extra TShockAPI.dll in the main folder on my install. I resolved that.

Now I'm seeing a different problem. Now I get a message about accounts existing when I know they do not exist. In fact, after it gives me the message, I can immediately follow with a /register charname to register the account successfully with the charname as the password.
